src/main/java/com/advent/of/code/day4/Board.java

```diff
@@ -0,0 +1,7 @@
+package com.advent.of.code.day4;
+
+interface Board {
```

Comment:
ten interfejs jest tu oczywiście do niczego nie potrzebny - w świecie EJB/Springa interfejsy są bardzo ważne (konkretnie te frameworki potrafią automatycznie wygenerować pewien kod w oparciu o zadeklarowany interfejs, bez udziału programisty) i programiści, którzy mają takie naleciałości bardzo często tworzą "dla sztuki" interfejs dla każdej swojej klasy co produkuje czasem takie potworki jak np interfejs "MyLogicService" i klasę "MyLogicServiceImpl" - brrr :D 

Interfejsy są jedną z najlepszych rzeczy w Javie ale ja polecam ostrożne podchodzenie do tematu - zasada KISS i YAGNI tutaj powinna mieć mocne zastosowanie. Ja sobie stworzyłem ten interfejs bo się spodziewałem, że w drugiej części zadania zmienią się zasady gry (wtedy stworzyłbym sobie kolejną implementację Boarda - np ExtendedBoard) ale się pomyliłem a kodu nie zrefaktorowałem :) Ale chociaż jest okazja o tym wspomnieć

Reply:
Właśnie się zastanawiałem jaki jest konkretnie zamysł **Board** i **DefaultBoard**, skoro można byłoby to wszystko zmieścić w jednej klasie. Fajnie, że wytłumaczyłeś :)